default/etc/e-smith/templates/httpd/proxy-extra/30ProxyPass

```diff
@@ -1,5 +1,33 @@
     # Reverse Proxy (with exclusion of local Letsencrypt challenge path)
     ProxyPassMatch ^/.well-known/acme-challenge/ !
+{
+    if ($WebSockets eq 'enabled' ) {
+        my $ws = $Target;
+        $ws =~ s/(http|https):\/+//;
+
+        my $path;
+        my $url;
+        if ($WebSocketsPath eq '') {
+            $path = '.*';
+            $url = '%\{REQUEST_URI\}';
+        } else {
+            $path = $WebSocketsPath;
+            $url = $WebSocketsPath;
+        }
+
+        $OUT .= qq(
+    # Websockets proxypass
+    RewriteCond \%\{HTTP:Upgrade\} websocket [NC]
+    RewriteCond \%\{HTTP:Connection\} upgrade [NC]
+);
+        if ($Target =~ /https:/) {
+            $OUT .= "    RewriteRule $path wss://".$ws.$url." [P,L]";
+        } elsif ($Target =~ /http:/) {
+            $OUT .= "    RewriteRule $path ws://".$ws.$url." [P,L]";
+        }
```

Comment:
Complement the previous suggestion. The "s" for SSL in URI scheme is implied by the http=>ws substitution above.

```suggestion
    $OUT .= "    RewriteRule ${path} ${ws}${url} [P,L]";
```